packages/deployments/contracts/contracts/core/connext/helpers/RelayerProxyHub.sol

```
@@ -81,13 +253,68 @@ contract RelayerProxyHub is RelayerProxy {
+    require(hubConnectors[fromChain] != address(0), "No hub connector");
+
+    if (fromChain == 100 || fromChain == 10200) {
+      IGnosisHubConnector.GnosisRootMessageData memory data = abi.decode(
+        encodedData,
+        (IGnosisHubConnector.GnosisRootMessageData)
+      );
+      IGnosisHubConnector(hubConnectors[fromChain]).executeSignatures(data._data, data._signatures);
+    }
+
+    if (fromChain == 42161 || fromChain == 421613) {
+      IArbitrumHubConnector.ArbitrumRootMessageData memory data = abi.decode(
+        encodedData,
+        (IArbitrumHubConnector.ArbitrumRootMessageData)
+      );
+      IArbitrumHubConnector(hubConnectors[fromChain]).processMessageFromRoot(
+        data._nodeNum,
+        data._sendRoot,
+        data._blockHash,
+        data._proof,
+        data._index,
+        data._message
+      );
+    }
+
+    if (fromChain == 10 || fromChain == 420) {
+      IOptimismHubConnector.OptimismRootMessageData memory data = abi.decode(
+        encodedData,
+        (IOptimismHubConnector.OptimismRootMessageData)
+      );
+      IOptimismHubConnector(hubConnectors[fromChain]).processMessageFromRoot(
+        data._target,
+        data._sender,
+        data._message,
+        data._messageNonce,
+        data._proof
+      );
+    }
+
+    if (fromChain == 324 || fromChain == 280) {
+      IZkSyncHubConnector.ZkSyncRootMessageData memory data = abi.decode(
+        encodedData,
+        (IZkSyncHubConnector.ZkSyncRootMessageData)
+      );
+      IZkSyncHubConnector(hubConnectors[fromChain]).processMessageFromRoot(
+        data._l2BlockNumber,
+        data._l2MessageIndex,
+        data._l2TxNumberInBlock,
+        data._message,
+        data._proof
+      );
+    }
+
+    if (fromChain == 137 || fromChain == 80001) {
+      IPolygonHubConnector(hubConnectors[fromChain]).receiveMessage(encodedData);
+    }
+
+    processedRootMessages[fromChain][l2Hash] = true;
```

I'd probably move this all the way up after the checks, just to follow the cei pattern. Also, is it correct to assume none of the functions called in this contract fail silently? As in, they return instead of reverting?